### DISCUSSION TITLE
Memory window 'disconnects' between debugging sessions

### DISCUSSION THREAD

**Discussion by Yazwh0** [microsoft/vscode-discussions#866] [Extension Development QnA] [answered]
I'm writing a debugger for an emulator, so the ability to view the contents of the emulated machine's memory is great.

The debugger uses DAP and the memory is exposed via a 'variable' at runtime (ie MemoryReference is set), and as you'd expect it updates as the memory changes.

However if you stop debugging and start again the window is unresponsive. It has to be closed and reopened, which is a bit annoying for the user.

Is there a way to convince VSCode to 'reattach' the memory window, or a way to define that it should work through a debugger restart?

**Comment by isidorn** [comment, marked as answer]
Sounds like a VS Code bug. I think the best would be to file an issue here https://github.com/microsoft/vscode/issues
I think even a PR would be welcome for this. Code pointer https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/browser/debugMemory.ts#L17

**Reply by Yazwh0** [reply]
If I knew Typescript! Not sure debugging VSCode is a good first introduction!
